### ISSUE TITLE
Login to Paperless-Ngx with Swift Paperless 

### ISSUE THREAD

**Issue by newcoder50** [paulgessinger/swift-paperless#157]
### App version

1.5.1

### Paperless-ngx backend version

2.13.5

### Description

I use Paperless-Ngx just local without connection from the Internet. Now when i want use the Swift-Paperless to login to Paperless i must type the url link but when i type my local ip-adress for paperless, i can't login to paperless. i have a picture 
[Unbenannttt.pdf](https://github.com/user-attachments/files/17893589/Unbenannttt.pdf)
[logs.txt](https://github.com/user-attachments/files/17893594/logs.txt)

**Comment by paulgessinger** [comment]
Hey @newcoder50,

Can you try putting `http://` in front of the IP explicitly? The app by default tries to use HTTPS which is likely causing the error for you!

**Comment by newcoder50** [comment]
Thank you

**Comment by newcoder50** [comment]
> Hey @newcoder50,
> 
> Can you try putting `http://` in front of the IP explicitly? The app by default tries to use HTTPS which is likely causing the error for you!

Thank you

**Comment by stefmatik** [comment]
Thanks for the awesome app!

I don't know if it's okay to hijack another issue topic, but I'm also facing a login issue.
Unfortunately, I have http in front, but I can't connect. I checked with an Android app and it's working there with the same IP and same username + password.
This is my log (I might have tested an https url behind a reverse proxy and the http ip request):
```
[24.11.2024, 19:57] [General] App becomes active
[24.11.2024, 19:57] [General] Checking backend URL <private>
[24.11.2024, 19:57] [General] Derive URL: <private> + <private> -> <private>
[24.11.2024, 19:57] [API] Headers for check request: <private>
[24.11.2024, 19:57] [General] Checking valid-looking URL <private>
[24.11.2024, 19:57] [General] Did not get identities from keychain
[24.11.2024, 19:57] [General] Checking API error: Error Domain=NSURLErrorDomain Code=-1003 UserInfo={_kCFStreamErrorCodeKey=8, NSUnderlyingError=0x30058eaf0 {Error Domain=kCFErrorDomainCFNetwork Code=-1003 UserInfo={_kCFStreamErrorDomainKey=12, _kCFStreamErrorCodeKey=8, _NSURLErrorNWResolutionReportKey=<private>, _NSURLErrorNWPathKey=satisfied (Path is satisfied), interface: en0[802.11], ipv4, dns, uses wifi}}, _NSURLErrorFailingURLSessionTaskErrorKey=<private>, _NSURLErrorRelatedURLSessionTaskErrorKey=<private>, NSLocalizedDescription=<private>, NSErrorFailingURLStringKey=<private>, NSErrorFailingURLKey=<private>, _kCFStreamErrorDomainKey=12}
[24.11.2024, 19:57] [General] Checking backend URL <private>
[24.11.2024, 19:57] [General] Value is empty
[24.11.2024, 19:57] [General] Checking backend URL <private>
[24.11.2024, 19:57] [General] Derive URL: <private> + <private> -> <private>
[24.11.2024, 19:57] [API] Headers for check request: <private>
[24.11.2024, 19:57] [General] Checking valid-looking URL <private>
[24.11.2024, 19:57] [General] Did not get identities from keychain
[24.11.2024, 19:57] [General] Attempting login with url <private>
[24.11.2024, 19:57] [General] Derive URL: <private> + <private> -> <private>
[24.11.2024, 19:57] [General] Sending login request with headers: <private>
[24.11.2024, 19:57] [General] Did not get identities from keychain
[24.11.2024, 19:57] [General] Token request response was not 200 but 403, <private>
[24.11.2024, 19:58] [General] Attempting login with url <private>
[24.11.2024, 19:58] [General] Derive URL: <private> + <private> -> <private>
[24.11.2024, 19:58] [General] Sending login request with headers: <private>
[24.11.2024, 19:58] [General] Did not get identities from keychain
[24.11.2024, 19:58] [General] Token request response was not 200 but 403, <private>
[24.11.2024, 19:58] [General] Something went wrong trying to find the idenities in the keychain
```

Yesterday in my tests, I was able to connect. Is it possible, that is has to do with `PAPERLESS_URL=<url>` being set?

**Comment by paulgessinger** [comment]
Hey @stefmatik, sure!

The log seems to indicate that the backend somehow does not accept the credentials (username and password) although they are provided. 

I guess you already double-checked that there's no typo? Other than that, do you use any kind of reverse proxy that might be messing with the request?

**Comment by stefmatik** [comment]
I'm sure that there is no typo, since I tried it like 10 times.
I just tried the Testflight version and trying with credentials didn't work but without credentials did (I have autologin activated).
So I guess that was the error, but I didn't find a way to try without credentials in the App Store app.
Thanks for your help!

**Comment by paulgessinger** [comment]
@stefmatik Ah, ok. I guess it's possible this doesn't play well. Can you share a few more details how the autologin is configured? Maybe I can reproduce it that way.

**Comment by stefmatik** [comment]
@paulgessinger I have set `PAPERLESS_AUTO_LOGIN_USERNAME` to a username.

**Comment by paulgessinger** [comment]
I added a hint to failing credentials pointing at the login possibility without credentials.

In my testing, I get a CSRF error, which is weird. I don't know how other apps would get any other results, since the CSRF error would be fully server side. Aside from this, the CSRF error persists even after I turned the setting off again. I asked about this on the Matrix channel, let's see if I get a reply.
